### PR TITLE
improve: inherit node default gateway if no routable bgp peer exist

### PR DESCRIPTION
**Pull request by hhyasdf**
Pull Request Description
---

### Special notes for reviews

### CODE REVIEW

File: pkg/daemon/controller/utils.go [hunk 1]
```diff
@@ -270,7 +270,8 @@ func nodeBelongsToNetwork(nodeName string, network *networkingv1.Network) bool {
 }
 
 func collectGlobalNetworkInfoAndInit(ctx context.Context, client client.Reader, nodeVxlanIfName, nodeName string,
-	bgpManager *bgp.Manager, recordBGPPeers bool) (vxlanForwardNodeIfName string, bgpGatewayIP net.IP, err error) {
+	bgpManager *bgp.Manager, recordBGPPeers bool) (vxlanForwardNodeIfName string, attachedBGPNetworkExist bool,
```
Comment (mars1024) [inline pkg/daemon/controller/utils.go:273]: the return "attachedBGPNetworkExist" may always be false

File: pkg/daemon/route/utils.go [hunk 2]
```diff
@@ -402,19 +402,52 @@ func ensureRoutesForVlanSubnet(forwardLink netlink.Link, cidr *net.IPNet, gatewa
...
+	}, netlink.RT_FILTER_TABLE)
+	if err != nil {
+		return fmt.Errorf("failed to list route for table %v: %v", table, err)
+	}
+
+	for _, route := range routeList {
+		// cannot use route.Equal() because of empty fields
+		if daemonutils.IsDefaultRoute(&route, family) &&
+			// TODO: support multiple bgp gateway
+			(!route.Gw.Equal(defaultRoute.Gw) || route.LinkIndex != defaultRoute.LinkIndex) {
```
Comment (mars1024) [inline pkg/daemon/route/utils.go:444]: can we restrict the deletion scope here, in other words, is the judgement for the route deletion safe enough?

File: pkg/daemon/controller/utils.go (hunk 1 above)
Comment (hhyasdf) [reply pkg/daemon/controller/utils.go:273]: fixed

File: pkg/daemon/route/utils.go (hunk 2 above)
Comment (hhyasdf) [reply pkg/daemon/route/utils.go:444]: Yes, it is safe, only daefault routes with different gateway and interface will be deleted.

**Review by mars1024 — APPROVED** [review]
/lgtm

**Pull request merged**